pkg/splunk/enterprise/util_test.go

```diff
@@ -260,4 +270,85 @@ func TestRemoveOwenerReferencesForSecretObjectsReferredBySmartstoreVolumes(t *te
 	if err != nil {
 		t.Errorf("Couldn't Remove S3 Secret object references %v", err)
 	}
+
+	// If the secret object doesn't exist, should return an error
+	// Here in the volume references, secrets splunk-test-sec_1, to splunk-test-sec_4 doesn't exist
+	cr = enterprisev1.ClusterMaster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "idxCluster",
+			Namespace: "testWithNoSecret",
+		},
+		Spec: enterprisev1.ClusterMasterSpec{
+			SmartStore: enterprisev1.SmartStoreSpec{
+				VolList: []enterprisev1.VolumeSpec{
+					{Name: "msos_s2s3_vol", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-sec_1"},
+					{Name: "msos_s2s3_vol_2", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-sec_2"},
+					{Name: "msos_s2s3_vol_3", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-sec_3"},
+					{Name: "msos_s2s3_vol_4", Endpoint: "https://s3-eu-west-2.amazonaws.com", Path: "testbucket-rs-london", SecretRef: "splunk-test-sec_4"},
+				},
+			},
+		},
+	}
+
+	// S3 secret owner reference removal, with non-existing secret objects
+	err = DeleteOwnerReferencesForS3SecretObjects(client, secret, &cr.Spec.SmartStore)
+	if err == nil {
+		t.Errorf("Should report an error, when the secret object referenced in the volume config doesn't exist")
```

Comment:
The error message can be "Error in owner ref removal as SecretRef %s doesn't exist"

Reply:
Here, we are expecting an error, but if the test doesn't return an error, that is when we are printing this message. So, I guess, the message in the current form is reasonable.